core/aws-core/src/main/java/software/amazon/awssdk/awscore/internal/AwsExecutionContextBuilder.java

```diff
@@ -121,13 +123,40 @@ private AwsExecutionContextBuilder() {
         interceptorContext = runInitialInterceptors(interceptorContext, executionAttributes, executionInterceptorChain);
 
         Signer signer = null;
-        if (isAuthenticatedRequest(executionAttributes)) {
-            AuthorizationStrategyFactory authorizationStrategyFactory =
-                new AuthorizationStrategyFactory(interceptorContext.request(), metricCollector, clientConfig);
-            AuthorizationStrategy authorizationStrategy =
-                authorizationStrategyFactory.strategyFor(executionParams.credentialType());
-            authorizationStrategy.addCredentialsToExecutionAttributes(executionAttributes);
-            signer = authorizationStrategy.resolveSigner();
+
+        // Note, not setting IS_NONE_AUTH_TYPE_REQUEST in newly generated clients, makes earlier logic not sufficient, as
+        // NoneAuthTypeRequestTest fails.
+        // It is easiest to keep NoneAuthTypeRequest passing by continuing to set IS_NONE_AUTH_TYPE_REQUEST in newly generated
+        // clients. This might become odd in the future, when authType=none is deprecated (authType itself will be less favored
+        // in lieu of authList, to support multiple auth schemes). But we can cross that bridge as part of those changes?
```

Comment:
I'm not very sure if we want to go the direction of this PR v/s continue to set this attribute for now, i.e., discard this PR. No tests are currently broken, if we continue setting IS_NONE_AUTH_TYPE_REQUEST.

I thought it was natural to try to remove the setter (as explained a bit in the code comments in these lines), so started with that, but needed add the conditional branching logic below to make NoneAuthTypeRequestTest pass. My nervous ness with adding these conditions comes from hoping that I've covered all cases.

Please review this question before reviewing lower level code feedback.

Reply:
Also, it seems this branching logic (in aws-core) has some relation with the `shouldDoSraSigning()` in SigningStage (sdk-core). I think that's ok, but wanted to bring attention to that for your consideration too.

Reply:
Agreed offline on keeping this attribute for backwards compatibility reasons for now.

Reply:
I'm not sure I completely understand 
> It is easiest to keep NoneAuthTypeRequest passing by continuing to set IS_NONE_AUTH_TYPE_REQUEST in newly generated clients.
Can we make a note to add compatibility tests for this feature in the cross-module testing that we're presumably planning?

Reply:
Additionally, can we remove these commented lines and if needed, add to one of the quip docs for the project as implementation notes etc.

Reply:
yup, added notes for testing and in implementation notes.